src/components/Form.js

```diff
@@ -0,0 +1,182 @@
+import { useForm } from 'react-hook-form'
+import PropTypes from 'prop-types'
+import InputField from './InputField'
+import Button from './Button'
+import styled from 'styled-components'
+import { useState, useRef } from 'react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+import React from 'react'
+
+Form.propTypes = {
+  onSubmit: PropTypes.func,
+}
+
+const schema = yup.object().shape({
+  name: yup.string().required(),
+  phone: yup.number().positive().integer().required(),
+  email: yup.string().required(),
+  department: yup.string().required(),
+  skill1: yup.string().required(),
+  skill2: yup.string(),
+  skill3: yup.string(),
+  skill4: yup.string(),
+  skill5: yup.string(),
+  skill6: yup.string(),
+})
+
+export default function Form({ onSubmit }) {
+  const ref = useRef(null)
+  const {
+    register,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(schema) })
+  const [isDisabled, setIsDisabled] = useState(true)
+
+  return (
+    <StyledForm
+      onSubmit={handleSubmit}
+      onChange={validateForm}
+      aria-label="create a new user"
+      role="form"
+      ref={ref}
+    >
+      <InputField
+        label="name:"
+        role="input"
+        {...register('name', {
+          required: true,
+          maxLength: 20,
+          pattern: /^[A-Za-z]+$/i,
+        })}
+      />
+      {errors.name && 'Name is required'}
```

Comment:
Maybe add a styled error component which is in a different color or something like that and display the error messages with that.
```suggestion
      {errors.name && <Error>Name is required</Error>}
```